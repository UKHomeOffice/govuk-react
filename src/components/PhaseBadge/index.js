// https://govuk-elements.herokuapp.com/alpha-beta-banners/

import glamorous from "glamorous";
import React from "react";

import { BLUE, WHITE } from "govuk-colours";
import {
  FONT_SIZE,
  LINE_HEIGHT,
  MEDIA_QUERIES,
  NTA_LIGHT
} from "../../constants/index";

const GPhaseBadge = glamorous.strong({
  display: "inline-block",
  margin: "0 8px 0 0",
  padding: "2px 5px",
  fontFamily: NTA_LIGHT,
  fontWeight: 700,
  fontSize: FONT_SIZE.SIZE_14,
  lineHeight: LINE_HEIGHT.SIZE_14,
  textTransform: "uppercase",
  letterSpacing: "1px",
  textDecoration: "none",
  color: WHITE,
  backgroundColor: BLUE,
  WebkitFontMmoothing: "antialiased",
  MozOsxFontSmoothing: "grayscale",
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: FONT_SIZE.SIZE_16,
    lineHeight: LINE_HEIGHT.SIZE_16,
    paddingBottom: 0
  }
});

const PhaseBadge = (props) => <GPhaseBadge {...props} />;

export default PhaseBadge;
