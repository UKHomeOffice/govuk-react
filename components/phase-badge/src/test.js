import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import PhaseBadge from './';

expect.addSnapshotSerializer(createSerializer(emotion));

const wrapper = <PhaseBadge>beta</PhaseBadge>;

describe(PhaseBadge, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
