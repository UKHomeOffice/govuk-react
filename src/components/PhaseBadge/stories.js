import React from "react";
import { storiesOf } from "@storybook/react";

import PhaseBadge from ".";

storiesOf("PhaseBadge", module).add("PhaseBadge", () => (
  <PhaseBadge>beta</PhaseBadge>
));
