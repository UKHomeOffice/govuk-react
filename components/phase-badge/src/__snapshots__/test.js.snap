// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseBadge matches wrapper snapshot: wrapper mount 1`] = `
.c1 {
  display: inline-block;
  margin: 0 8px 0 0;
  padding: 2px 5px;
  font-family: "nta",Arial,sans-serif;
  font-weight: 700;
  font-size: 14px;
  line-height: 1.1428571429;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: #005ea5;
  -webkit-font-mmoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0 {
  margin-bottom: 0;
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 16px;
    line-height: 1.25;
    padding-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Styled(PhaseBadge)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            [Function],
          ],
        },
        "displayName": "Styled(PhaseBadge)",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "margin": [Function],
          "mb": [Function],
          "padding": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
  >
    <PhaseBadge
      className="c0"
    >
      <styled.strong
        className="c0"
      >
        <StyledComponent
          className="c0"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "display: inline-block; margin: 0 8px 0 0; padding: 2px 5px; font-family: \\"nta\\", Arial, sans-serif; font-weight: 700; font-size: 14px; line-height: 1.1428571429; text-transform: uppercase; letter-spacing: 1px; text-decoration: none; color: #ffffff; background-color: #005ea5; -webkit-font-mmoothing: antialiased; -moz-osx-font-smoothing: grayscale; @media only screen and (min-width: 641px) {
  font-size: 16px; line-height: 1.25; padding-bottom: 0;
}",
                ],
              },
              "displayName": "styled.strong",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "strong",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <strong
            className="c0 c1"
          >
            beta
          </strong>
        </StyledComponent>
      </styled.strong>
    </PhaseBadge>
  </StyledComponent>
</Styled(PhaseBadge)>
`;
